.github/workflows/check-xpack-packetbeat.yml

```diff
@@ -0,0 +1,29 @@
+name: check-x-pack-packetbeat
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/check-xpack-packetbeat.yml'
+      - 'x-pack/packetbeat/**'
+      - 'packetbeat/**'
+
+env:
+  BEAT_MODULE: 'x-pack/packetbeat'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Fetch Go version from .go-version
+      run: echo "GO_VERSION=$(cat .go-version)" >> $GITHUB_ENV
+    - uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: Install libpcap-dev
+      run: sudo apt-get install -y libpcap-dev
+    - name: Run check/update
+      uses: magefile/mage-action@v2
+      with:
+        args: check update
```

Comment:
Is there a reason why the list of checks changed?

This new Github workflow runs just `mage check update` and before in Jenkins it ran:

```sh
# what you have here
mage check;
mage update;
# additionally from the Jenkins file
make -C packetbeat check;
make -C packetbeat update;
make check-no-changes;
```

Reply:
See my previous comment -> https://github.com/elastic/beats/pull/32711#discussion_r947800792

Reply:
@v1v sorry I don't understand how your comment would answer my question.

My question was "why the checks in new workflows differ from what we had before in the Jenkins files" not "why do we need 2 GitHub workflows".

Reply:
> Is there a reason why the list of checks changed?

They have not changed, checks are running the same steps since they have split in two different actions:
1) the one for `packetbeat` which uses `make`
2) the one for `x-pack/packetbeat` which uses `mage`

Those GitHub actions run **always** when a change happens in either `packetbeat` or `x-pack/packetbeat`.

Hence, `check-x-pack-packetbeat.yml` runs
```
# what you have here
mage check;
mage update;
```

 while `check-packetbeat.yml` runs 

```
make -C packetbeat check;
make -C packetbeat update;
make check-no-changes;
```

The existing changes in Jenkins were originally done in https://github.com/elastic/beats/pull/23378, with a similar approach, but changed afterwards for simplicity in the `when` condition as stated in https://github.com/elastic/beats/pull/23695.

Reply:
> They have not changed, checks are running the same steps since they have split in two different actions:
> 1. the one for packetbeat which uses make
> 2. the one for x-pack/packetbeat which uses mage

Are you sure? Because from what I see, it ran both `mage` **and** `make` for both `packetbeat` and `x-pack/packetbeat`:

<img width="582" alt="Screenshot 2022-08-18 at 15 55 37" src="https://user-images.githubusercontent.com/887952/185412846-75af6366-fcfa-4629-ac7b-8855111e5c38.png">
<img width="586" alt="Screenshot 2022-08-18 at 15 54 29" src="https://user-images.githubusercontent.com/887952/185412551-49f8fa0f-fe48-41f9-b47f-d8cc836c5fe5.png">

So, when I looked at the diff it caused my initial question in this thread.

I might be missing something here, I don't mean to be annoying but it's better to clarify this than miss some checks in the future.

Reply:
If you see:
1) https://github.com/elastic/beats/pull/32711/files#r949218960 you will see it will run the make commands.
2) https://github.com/elastic/beats/pull/32711/files#r949219714 you will see it will run the mage commands.

Therefore those are equivalent what it ran in Jenkins. right?

Reply:
@v1v how come that we see different things looking at the same diff?

## packetbeat

was (packetbeat/Jenkinsfile.yml)

```
make -C packetbeat check;
make -C packetbeat update;
make check-no-changes;
cd x-pack/packetbeat;
mage check;
mage update;
```

runs make AND mage targets.

is now (.github/workflows/check-packetbeat.yml)

```
export PATH=$PATH:$(go env GOPATH)/bin
go install github.com/magefile/mage@latest
make -C ${{ env.BEAT_MODULE }} check update
make check-no-changes
```

`.github/workflows/check-packetbeat.yml` installs mage but does not run any targets with mage, it's running only make targets.

## x-pack/packetbeat

was (x-pack/packetbeat/Jenkinsfile.yml)

```
mage check;
mage update;
make -C packetbeat check;
make -C packetbeat update;
make check-no-changes;
```

runs mage AND then make targets.

is now (.github/workflows/check-xpack-packetbeat.yml)

```
mage check update
```

it runs only the mage action with `check update` as arguments.

Reply:
I get it now, so the union of these 2 new Github workflows is supposed to substitute each former check. So, basically instead of each Jenkins file we'll run both Github actions and they together will run:

```
export PATH=$PATH:$(go env GOPATH)/bin
go install github.com/magefile/mage@latest
make -C ${{ env.BEAT_MODULE }} check update
make check-no-changes
mage check update
```

which covers all the comments in each former Jenkins file.